src/steamship/data/plugin_instance.py

```diff
@@ -113,7 +113,7 @@ def delete(self) -> PluginInstance:
         req = DeleteRequest(id=self.id)
         return self.client.post("plugin/instance/delete", payload=req, expect=PluginInstance)
 
-    def train(self, training_request: TrainingParameterPluginInput) -> TrainPluginOutput:
+    def train(self, training_request: TrainingParameterPluginInput) -> Task[TrainPluginOutput]:
```

Comment:
Oh this is an interesting question. 

At the "client" level, we basically created a facade in which all methods return either `T` or `Task[T]`.

But we didn't go through and update the Plugin SDK for this as well, which we could have.

I'm pretty sure the response type for this ought to be:

```
Union[T, InvocableResponse[T]]
```

So in this case:

```
Union[TrainPluginOutput, InvocableResponse[TrainPluginOutput]]
```

If the training is still in progress, the plugin returns:

```
InvocableResponse(
  status=Task(stats=TaskState.running)
)
```

Reply:
But this IS the client side of the facade, not what the plugin SDK can actually return?  This how the client calls train, not how the plugin responds.

Reply:
Oh snap, I missed that. Then yes -- def `Task[T]`!!